### PR TITLE
Harvester boilerplate

### PR DESCRIPTION
**Pull request by mchurichi**
**Pull Request check list**

- [ ] Proper tests/regressions included?
- [X] Documentation updated?

**Affected functionality**
N/A

**Description of change**
The idea of this PR is to start outlining the general modular architecture of the Harvester (the Server should be in the same line). All components have a very minimal interface and a weak implementation that will change as soon as we start tackling them one by one.

The components have been shaped into different packages with well-defined interfaces. A central component called Manager takes care of the initialization and hydrates the components catalog. Some of the components must run in the background, those implement the Runnable interface. The Controller (which is a Runnable) will query from the other components (brokers), pulling and pushing stuff as needed.

**Which issue this PR fixes**
N/A

### CODE REVIEW

**Review by n-th — APPROVED** [review]
🔥

**Review by n-th — DISMISSED** [review]

File: pkg/common/log.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package common
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+var (
```
Comment (mgbcaio) [inline pkg/common/log.go:10]: Wouldn't this be better to be a `const` ?

File: cmd/harvester/cli/cli.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	}
+}
+
+func (c *HarvesterCLI) Run(args []string) int {
+	if len(args) != 1 {
+		c.logger.Error("Unknown arguments", args)
+		return 1
+	}
+
+	cfg, err := config.LoadFromDisk("conf/harvester/harvester.conf")
```
Comment (mgbcaio) [inline cmd/harvester/cli/cli.go:27]: Maybe move this path inside a `const` ? We can also do it later :)

**Review by mgbcaio — APPROVED** [review]
Just left some comments. Good work, Maxi! 💯

File: pkg/common/log.go (hunk 1 above)
Comment (mchurichi) [reply pkg/common/log.go:10]: good catch

**Review by mgbcaio — APPROVED** [review]
🦾

**Pull request merged**